clash-prelude/src/Clash/Sized/Vector.hs

```diff
@@ -1111,6 +1112,11 @@ scanl f z xs = ws
     ws = z `Cons` zipWith (flip f) xs (init ws)
 {-# INLINE scanl #-}
 
+-- | 'scanl' with no seed value
+scanl1 :: KnownNat n => (a -> a -> a) -> Vec (n+1) a -> Vec (n+1) a
+scanl1 op v = scanl op (head v) (tail v)
```

Comment:
Can't you do `(v :> vs)` or ``v `Cons` vs``? Or does that trip up the exhaustiveness checker?

Reply:
`Cons` should _never_ trip up the exhaustiveness check I think, the only reason `:>` does is because of nasty interaction between GADTs and pattern synonyms

Reply:
Although maybe pre GHC 8.10 it might get stupid, there were some meaningful changes then. I always forget we support GHCs older than I use locally

Reply:
That works!